src/main/java/org/adridadou/ethereum/EthereumProxy.java

```diff
@@ -184,7 +184,9 @@ public EthValue getBalance(EthAddress address) {
         return ethereum.getBalance(address);
     }
 
-    private void increasePendingTransactionCounter(EthAddress address) {
-        pendingTransactions.put(address, pendingTransactions.getOrDefault(address, BigInteger.ZERO).add(BigInteger.ONE));
+    private void increasePendingTransactionCounter(EthAddress address, EthHash hash) {
+        Set<EthHash> hashes = pendingTransactions.getOrDefault(address, new HashSet<EthHash>());
```

Comment:
ust to be safe, I think that using Collections.synchronizedSet(Set set) to avoid having issues would be a good idea.

Also, to avoid concurrent issues, I would change this code to:

Set hashes = pendingTransactions.computeifAbsent(address, (address) -> Collections.synchronizedSet(new HashSet<>()));

//do what you have to do

This way you avoid overriding the hashSet